geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java

```diff
@@ -721,12 +721,19 @@ public void addListener(PersistentStateListener listener) {
   @Override
   public void removeListener(PersistentStateListener listener) {
     synchronized (this) {
+      if (persistentStateListeners.isEmpty()) {
```

Comment:
even better would be:
`if (!persistentStateListeners.contains(listener)) { return; }`

Reply:
Thanks for comment. Updated.